### ISSUE TITLE
Remove "IR 1 D1"/etc from match report

### ISSUE THREAD

**Issue by rythos42** [TheNAF/naflm#66]
This was requested by a user, and it would be nice to have some discussion if possible.

There are fields "IR 1 D1	IR 1 D2	IR 2 D1	IR 2 D2	IR 3 D1	IR 3 D2" in the match report that _appear_ to have no use.

These are actually used to report on what *dice* the player rolled for their skill-up. If they are provided, the system narrows the list of possible skills to regular or double or MA, etc. However, as noted by the user, there is no way to tell this and the feature may be little used.

It could be worth looking at the database to see if anyone has ever set this in NAFLM. Personally I have no problem removing these fields, as they are the definition of feature-creep (not super useful, and in your face), but as I'm just the latest in a series of project "owners", I'd hate to remove something that someone loved!

**Comment by Shteve0** [comment]
We do need a way of submitting non-standard improvement rolls. By default
coaches can only select "normal" skills. Can we think of a more elegant
solution?  Maybe just titling that block "improvement rolls"?

**Comment by mfranchetti** [comment]
Certainly within my league they are used. I like it as a check on coaches selecting skills, as mistakes do happen. 
What I have thought would be an improvement would be to have them hidden by default and only shown when a player becomes eligible for a skill roll based on SPP. 
Should be possible with some JavaScript. 
On 3 Feb 2016 6:10 pm, Craig Fleming <notifications@github.com> wrote:This was requested by a user, and it would be nice to have some discussion if possible

There are fields "IR 1 D1   IR 1 D2 IR 2 D1 IR 2 D2 IR 3 D1 IR 3 D2" in the match report that appear to have no use

These are actually used to report on what dice the player rolled for their skill-up If they are provided, the system narrows the list of possible skills to regular or double or MA, etc However, as noted by the user, there is no way to tell this and the feature may be little used

It could be worth looking at the database to see if anyone has ever set this in NAFLM Personally I have no problem removing these fields, as they are the definition of feature-creep (not super useful, and in your face), but as I'm just the latest in a series of project "owners", I'd hate to remove something that someone loved!

**Comment by rythos42** [comment]
Updated the title!

I'm conflicted about what you said, Steve. On the one hand, I remember having this exact issue -- I couldn't select a double skill on a player. On the other hand, I just checked my local deployment and the skills list has *all* of the options in it for a player that has a skill-up. 

But if your league is using them, then definitely it should stay. I don't know about hiding/showing them based on skill-ups -- it feels like this would be a lot of coding just to have them appear in most match reports. (because after entering SPP, I feel like it's at least half of the games I play where a player gets a skill-up, so it would be shown). 

What about this as an option:

1. Remove IR2/IR3. There are loooong odds that these 4 drop-down lists would ever be used. 16SPP in a single game is your best chance of using both IR1 and IR2. *32SPP* in a single game is your best chance of using all 3. (because earlier levels are easier to get)
1. Rename IR1 D1 and IR2 D2 to "Improvement Rolls" as a single header above both columns.

How about that as an option?

**Comment by Shteve0** [comment]
Your list has them all, because you have an admin account ;)

**Comment by thefloppy1** [comment]
I would prefer to remove the dice rolling from the players, and make it server side.   In my league I think only 1 person so far has used this correctly.  that said, they also didn't use the Team Cas correctly either, but IR D1/D2 means nothing to anybody that doesn't understand what the titles are.

**Comment by hutchinsfairy** [comment]
+1 for keeping it in some form. We have been using it, and at least some of the coaches have appreciated being able to enter the dice rolls on game night then select from a limited list at their leisure.

Having said that, the current implementation is awkward at best. I think just renaming the headings would go a long way but if someone had some time to spare on it I would love to see the match report divided into "pre-match", "match" and "post-match" sections to make it more intuitive for the coaches.

If someone had a _lot_ of time... how about an input wizard? Logical steps for the coach to go through with dynamic prompts for things like fan factor, winnings and improvement rolls based on previously entered data! Also the actual moon, on some sort of stick device.

**Comment by mfranchetti** [comment]
Would it be more user friendly to record the result rather than the actual dice rolls?

Normal
Double
+MV/+AV Normal
+MV/+AV Double
+AG Normal
+ST Double

Recording the dice roll and converting it into the appropriate result is only useful where the user may not know the correct outcome. I would suggest that that is not a user case we're likely to encounter or need to support.
Additionally, where a player achieves a new skill, we can default to Normal so no additional input is required unless the result is one of the others. (1/12 cases)

**Comment by mfranchetti** [comment]
I've created a quick [demo](https://jsfiddle.net/mfranchetti/9vt1nrdw/) of my idea on dynamically showing the skill roll selectors. Let me know if I should try and integrate this into the current code.

**Comment by rythos42** [comment]
I think your demo is a _dramatic_ improvement over the existing feature. I'd say do it!

A small technical note -- I'm trying to use KnockoutJS wherever possible within NAFLM. If you're familiar with KO, and able to make your feature work with it that would be great! If not, it's not a big deal, I just find KO helps to make JS code more structured and organized.

**Comment by mfranchetti** [comment]
Haven't used KO but I'll take a look and see if I can get it to work. Are there any sample pages in OBBLM I can look at for your styling?

**Comment by rythos42** [comment]
Look at class_mobile_htmlout for my most usages. It's a JavaScript data binding framework, to keep the UI and behavior separated better.

**Comment by thefloppy1** [comment]
that is nice. I like that as a change.

**Comment by hutchinsfairy** [comment]
Looks great! Much cleaner. Whilst you're tweaking... is there any value to having the MVP as a dropdown? Seems like you could accommodate more weird and wonderful house rules with free input. Also, shouldn't the SPP box either be read-only or used as an override? Currently (in the demo) it can be set manually but gets reset.

**Comment by mfranchetti** [comment]
Yes, SPP will be a hidden field in the final implementation, it was only visible for the purposes of the demo. There'll probably be two hidden fields, pre-match SPP and post-match SPP, so that we can identify players who have crossed a threshold.
Not sure about the MVP, worth opening as a separate issue for discussion?
I'd also like to hide the **Aging** dropdowns too, but I will open that as a separate issue.
